operation/image_cropping.go

```diff
@@ -0,0 +1,134 @@
+package operation
+
+import (
+	"errors"
+	"image"
+	"strings"
+
+	"golang.org/x/image/draw"
+)
+
+// Define errors.
+var (
+	ErrInvalidCropBoundary          = errors.New("invalid crop boundary")
+	ErrCroppingAreaOutOfBound       = errors.New("crop boundary is outside of the original image")
+	ErrAlignmentMethoudNotSupported = errors.New("alignment method not supported")
+)
+
+// Defines a prototype for crop alignment function. (e.g. Center, TopLeft, BottomRight)
+// The first parameter is the boundary of the original image.
+// The second and third parameters are the width and height of the cropping area, respectively.
+type CropAlignment func(image.Rectangle, int, int) image.Rectangle
+
+func CenterAlignment(original_image_boundary image.Rectangle, cropped_width, cropped_height int) image.Rectangle {
```

Comment:
It's good practice to check the crop width and height are positive before proceeding with the cropping operation.
I think negative values could lead to unexpected behavior.

Reply:
I think so, a'll add some fix to the issue. : D